src/Symfony/Component/Runtime/Runner/FrankenPhpWorkerRunner.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime\Runner;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+use Symfony\Component\Runtime\RunnerInterface;
+
+/**
+ * A runner for FrankenPHP in worker mode.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+class FrankenPhpWorkerRunner implements RunnerInterface
+{
+    public function __construct(
+        private HttpKernelInterface $kernel,
+        private int $loopMax,
+    ) {
+    }
+
+    public function run(): int
+    {
+        // Prevent worker script termination when a client connection is interrupted
+        ignore_user_abort(true);
+
+        $server = array_filter($_SERVER, static fn (string $key) => !str_starts_with($key, 'HTTP_'), ARRAY_FILTER_USE_KEY);
+        $server['APP_RUNTIME_MODE'] = 'web=1&worker=1';
+
+        $handler = function () use ($server, &$sfRequest, &$sfResponse): void {
+            // Connect to the Xdebug client if it's available
+            if (\extension_loaded('xdebug') && \function_exists('xdebug_connect_to_client')) {
+                xdebug_connect_to_client();
+            }
+
+            // Merge the environment variables coming from DotEnv with the ones tied to the current request
+            $_SERVER += $server;
+
+            $sfRequest = Request::createFromGlobals();
+            $sfResponse = $this->kernel->handle($sfRequest);
+
+            $sfResponse->send();
+        };
+
+        $loops = 0;
+        do {
+            $ret = \frankenphp_handle_request($handler);
+
+            if ($this->kernel instanceof TerminableInterface && $sfRequest && $sfResponse) {
+                $this->kernel->terminate($sfRequest, $sfResponse);
+            }
+
+            gc_collect_cycles();
```

Comment:
I'm wondering about this: could it become a perf hog?

Reply:
Worker mode is relatively new to PHP so maybe there's something to improve on this topic in php-src itself?
Some new conditional trigger that'd be light to check when running the GC would be really useful?
Or maybe we can build something on gc_status?

If we want to give more control over this to end users, we could move the call into a kernel-terminate listener, that'd do it only in worker mode.

Reply:
It would be nice to move this logic to a listener and make it configurable, but maybe it could be done in a follow-up PR?